EIPS/eip-5075.md

```diff
@@ -0,0 +1,293 @@
+
+Note Transfers from specified addresses MAY skip the outflow and rate limit checks, with the addresses being able to be updated by owner or some other mechanism.
+  
+Contracts SHALL replace all ETH and Token based transfers with transferL where the transfers are to unknown external addresses or are publicly accessible.
+  
+```function transferL(address _to, uint256 _amount, address _token) public returns (bool success)```
+
+  
+**getLimitLeft**
+
+
+Returns the amount allowed within the rate limited transfer allowance for the current time with current outflow.
+
+```function getLimitLeft(address token) public view returns (uint256 rateLeft)```
+  
+  
+**whitelist**
+
+
+Allows an address or addresses to whitelist other addresses as msg.sender to not be rate limited.
+OPTIONAL - This allows for an owner or other mechanism to allow contracts such as YEARN to be able to move positions or manage positions without affecting or being affected by the rate limit and out flows. Use with caution and for contracts with known deployed bytecode and that have been verified to not be malicious or able to be maliciously used to allow for getting around the rate limit during hacks. Note remember to restrict the address able to use the whitelist and removeWhitelist fn.
+  
+```function whitelist(address addrs) public```
+
+```function removeWhitelist(address addrs) public```
+
+
+## Rationale
+By limiting token outflows at a per token level, malicious interactions can be limited in their damage as all net outflows are tracked and limited for all users based on a realtime allowance based on acceptable outflow relative to total amount held and within a given timeframe. The decision to track at the token level was for simplicity, gas efficiency and not needing to itself have vulnerabilities with oracles and accounting to track as a pool vs per asset. Whitelisting allows for contracts like yearn to be compatible if the destination is  a known address not likely to be used in an attack. Dynanmic rate limits will help not stop but limit the possible damage, as all outflows need to use a transfer fn to steal assets, which this stops if reaches the rate limit.
+
+
+## Backwards Compatibility
+This will be non backwards compatible with contracts that are NOT whitelisted which assume they can just withdraw large positions, if those positions are large relative to the target contracts net position by all users. Should this be the case the contract can be fully compatible by simply being whitelisted, skipping all rate limit checks. There should be no other problems with compatibility as uses the standard ERC20 interface for tokens and usual ether transfer to replace.
+
+
+## Reference Implementation
+
+```
+// SPDX-License-Identifier: CC0
+pragma solidity ^0.8;
+
+import"@openzeppelin/contracts/token/ERC20/IERC20.sol";
```

Comment:
```suggestion
import"IERC20.sol";
```
Implementing ERC20 is an exercise for the reader, we shouldn't be pinning to specific implementations.